### PR TITLE
Codify performance manifests

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Codify performance manifests to make it easier to introduce complex logic for applying this manifest.

- [x] cluster feature gate resource
- [x] kubelet config resource
- [x] machine config resource
- [x] machine config pool
- [x] tuned resource
- [x] add unittest for all generation methods

### CODE REVIEW

File: build/assets/scripts/pre-boot-tuning.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        ver=`awk '/version/ {print $2}' $line`
+        if [ "$ver" -gt "$current_ver" ]; then
+           current_ver=$ver
+           entry_file=$line
+        fi
+    done <<<$(egrep $(uname -r) -lr /boot/loader/entries/)
+
+    sed -i "s^initrd .*\$^& ${RHCOS_OSTREE_BOOTLOADER_PATH}iso_initrd.img^" $entry_file
+
+    #TODO - once RHCOS image contains the initrd content we can set parameters with rpm-ostree:
```
Comment (yanirq) [inline build/assets/scripts/pre-boot-tuning.sh]: We might actually be able to test that already with latest RHCOS changes , but ill do it in another PR

File: build/assets/tuned/openshift-node-network-latency [hunk 2]
```diff
@@ -0,0 +1,52 @@
+[main]
+summary=Openshift node optimized for deterministic performance at the cost of increased power consumption, focused on low latency network performance
+include=openshift-node
+
+[cpu]
+# https://github.com/redhat-performance/tuned/blob/master/profiles/latency-performance/tuned.conf
+# https://github.com/redhat-performance/tuned/blob/master/profiles/network-latency/tuned.conf
+force_latency=cstate.id:1|3
```
Comment (yanirq) [inline build/assets/tuned/openshift-node-network-latency]: there might be an issue with this format of setting latency  (using the OR operand) , investigating this but worth checking if tuned pod doesn't have any errors with that here .

File: pkg/controller/performanceprofile/components/machineconfig/machineconfig.go [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+	}
+
+	mc.Spec.Config = *ignitionConfig
+	mc.Spec.KernelArguments = getKernelArgs(profile.Spec.HugePages, profile.Spec.CPU.Isolated)
+
+	return mc, nil
+}
+
+func getKernelArgs(hugePages *performancev1alpha1.HugePages, isolatedCPUs *performancev1alpha1.CPUSet) []string {
+	kargs := []string{
```
Comment (yanirq) [inline pkg/controller/performanceprofile/components/machineconfig/machineconfig.go]: Can we pass these values in the CR as well ?  Can we and do we want to pass then in the CR ?

**Comment by yanirq** [comment]
@cynepco3hahue  overall looks ok ,  If there are specific files that were not migrated as they are and are worth a second look please specify them.
Also need to see e2e tests pass

File: pkg/controller/performanceprofile/components/machineconfig/machineconfig.go (hunk 3 above)
Comment (cynepco3hahue) [reply pkg/controller/performanceprofile/components/machineconfig/machineconfig.go]: Currently, I pass the whole performance profile.

File: build/assets/tuned/openshift-node-network-latency (hunk 2 above)
Comment (cynepco3hahue) [reply build/assets/tuned/openshift-node-network-latency]: We can fix it under the separate PR

**Comment by cynepco3hahue** [comment]
@yanirq e2e tests 

> @cynepco3hahue overall looks ok , If there are specific files that were not migrated as they are and are worth a second look please specify them.
> Also need to see e2e tests pass

Fixed unittest, e2e tests requires the deployment of the operator, will create following PR once #25  will be merged.

File: pkg/apis/performance/v1alpha1/performanceprofile_types.go [hunk 4]
```diff
@@ -44,15 +44,15 @@ type HugePages struct {
...
-	Count *int32 `json:"count,omitempty"`
+	Count int32 `json:"count,omitempty"`
 }
 
 // RealTimeKernel defines the set of parameters relevant for the real time kernel.
 type RealTimeKernel struct {
-	// Enabled enables real time kernel on relevant nodes.
-	Enabled *bool `json:"enabled,omitempty"`
+	// RepoURL defines the URL to the repository with real time kernel packages
+	RepoURL *string `json:"repoURL,omitempty"`
```
Comment (davidvossel) [inline pkg/apis/performance/v1alpha1/performanceprofile_types.go]: is this going to eventually turn back into a boolean soon?

File: pkg/controller/performanceprofile/components/tuned/tuned_test.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  - match:
+    - label: label1
+      match:
+      - label: label2
+        value: label2
+      value: label1
+`
+
+var _ = Describe("Tuned", func() {
+	Context("with worker real time kerbnel profile", func() {
```
Comment (davidvossel) [inline pkg/controller/performanceprofile/components/tuned/tuned_test.go]: kerbnel/kernel
